index.js

```diff
@@ -148,9 +173,9 @@ async function fastifyStatic (fastify, opts) {
 
           setUpHeadAndGet(routeOpts, route, `/${file}`, rootPath)
 
-          const key = path.posix.basename(route)
+          const key = path.basename(route)
           if (indexes.includes(key) && !indexDirs.has(key)) {
-            indexDirs.set(path.posix.dirname(route), rootPath)
```

Comment:
Sorry if this was asked before, but why this change? Seems like it might introduce unnecessary risk to this PR

Reply:
It does the same thing for the purposes of that part of the code, but you're right that it's an unnecessary change